core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/CurateOutcomeBuildItem.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.deployment.pkg.builditem;
+
+import io.quarkus.bootstrap.model.AppModel;
+import io.quarkus.bootstrap.resolver.AppModelResolver;
+import io.quarkus.builder.item.SimpleBuildItem;
+
+public final class CurateOutcomeBuildItem extends SimpleBuildItem {
+
+    private final AppModel effectiveModel;
+    private final AppModelResolver resolver;
+
+    public CurateOutcomeBuildItem(AppModel effectiveModel, AppModelResolver resolver) {
```

Comment:
It would be nice if we could have actual inputs for each element of the class path, similar to the application archives thing but for the whole class path, where you can tell whether it's a JAR or filesystem path.

I hacked up something like this for the native image extension experiment.  It's not really worth copying but it validates the idea a bit.

Reply:
This could definitely be improved, but I think we can do the initial change without it.